### PR TITLE
Re-organize algolia configs centralizing initialization, utils and cu…

### PR DESCRIPTION
**Pull request by mcstover**
…stom categories.

### CODE REVIEW

File: src/pages/Lend/Filter/FilterComponents/SelectedRefinements.vue [hunk 1]
```diff
@@ -66,10 +66,6 @@ export default {
 		ClearAllRefinements,
 	},
 	props: {
-		customCategories: {
```
Comment (mcstover) [inline src/pages/Lend/Filter/FilterComponents/SelectedRefinements.vue]: now provided by algolia-custom-categories-mixin

File: src/pages/Lend/Filter/FilterSections/FilterSectionCategories.vue [hunk 2]
```diff
@@ -26,10 +26,6 @@ export default {
 		algoliaCustomCategories,
 	],
 	props: {
-		customCategories: {
```
Comment (mcstover) [inline src/pages/Lend/Filter/FilterSections/FilterSectionCategories.vue]: now provided by algolia-custom-categories-mixin

File: src/pages/Lend/Filter/FilterSections/LocationMulti/FilterSectionLocationMulti.vue [hunk 3]
```diff
@@ -36,7 +36,7 @@ export default {
 		AisRefinementList,
 	},
 	props: {
-		locationLvl1Data: {
+		allLocationsLvl1: {
```
Comment (mcstover) [inline src/pages/Lend/Filter/FilterSections/LocationMulti/FilterSectionLocationMulti.vue]: renamed to better match pattern of other all facets data

File: src/pages/Search/SentenceSearch.vue [hunk 4]
```diff
@@ -158,17 +158,10 @@ import WwwPage from '@/components/WwwFrame/WwwPage';
 import LendHeader from '@/pages/Lend/LendHeader';
 // This mixin provides some algolia search instance initialization on mounted
 import algoliaInit from '@/plugins/algolia-init-mixin';
-// This mixin provides config for our indices + loan channel categories
-import algoliaConfig from '@/plugins/algolia-config-mixin';
```
Comment (mcstover) [inline src/pages/Search/SentenceSearch.vue]: no longer needed

File: src/plugins/algolia-custom-categories-mixin.js [hunk 5]
```diff
@@ -4,6 +4,155 @@ export default {
 	data() {
 		return {
 			customCategoryAttributes: ['sector.name', 'themeData.loanThemeTypeName', 'tags.name'],
+			/* eslint-disable max-len */
```
Comment (mcstover) [inline src/plugins/algolia-custom-categories-mixin.js]: moved here to centralize custom category datas

File: src/plugins/algolia-init-mixin.js [hunk 6]
```diff
@@ -1,151 +1,9 @@
-import _filter from 'lodash/filter';
 import _map from 'lodash/map';
 import algoliasearch from 'algoliasearch/lite';
 import { history as historyRouter } from 'instantsearch.js/es/lib/routers';
+import { stateToRoute, routeToState, fundraisingIndices } from '@/util/algoliaUtils';
 
-const indexKeys = [
```
Comment (mcstover) [inline src/plugins/algolia-init-mixin.js]: moved all of these into algoliaUtils

File: src/plugins/algolia-init-mixin.js [hunk 7]
```diff
@@ -181,6 +39,17 @@ export default {
 			allTagNames: [],
 		};
 	},
+	computed: {
```
Comment (mcstover) [inline src/plugins/algolia-init-mixin.js]: required for initialization in most instances so added here.

File: src/plugins/algolia-init-mixin.js [hunk 8]
```diff
@@ -199,85 +68,29 @@ export default {
...
-							label: facet.value,
-							value: facet.value,
-						};
-					});
-				}
-			});
-		},
-		setAllTags() {
+		// Fetch all facet values for a specific facet and merge them with the filtered facet values
+		setAllFacetValues(facetName, facetDestinationKey) {
```
Comment (mcstover) [inline src/plugins/algolia-init-mixin.js]: refactored for reuse

File: src/pages/Lend/Filter/FilterComponents/SelectedRefinements.vue (hunk 1 above)
Comment (BoulderBrains) [reply src/pages/Lend/Filter/FilterComponents/SelectedRefinements.vue]: 👍

File: src/pages/Lend/Filter/FilterComponents/LendFilterMenu.vue [hunk 9]
```diff
@@ -63,6 +63,8 @@ import {
 	AisStateResults,
 } from 'vue-instantsearch';
 // Custom Categories
+// import the following to access "customCategories"
+// import algoliaCustomCategories from '@/plugins/algolia-custom-categories-mixin';
```
Comment (ksekhri) [inline src/pages/Lend/Filter/FilterComponents/LendFilterMenu.vue]: Nice

File: src/plugins/algolia-init-mixin.js (hunk 6 above)
Comment (ksekhri) [reply src/plugins/algolia-init-mixin.js]: Makes sense

**Pull request merged**